From ba61288d7cc3d139a911c8fcd3a25941c20a07e4 Mon Sep 17 00:00:00 2001
From: Pavan Kumar S <pavan.kumar.s@intel.com>
Date: Fri, 8 May 2015 14:51:47 +0530
Subject: [PATCH] [WA][TYPE-C]: Update pmic reg for enabling host-mode

To enable the host mode functionality in typec, reverse boost needs
to be enabled from pmic. This is done through gpio.
Since BIOS hasn't exposed this as of now, this is a w/a patch to write directly.

Change-Id: Ia7f640553db1416bdb3bee3326e3234a731aba14
Tracked-On: https://jira01.devtools.intel.com/browse/IMINAN-35352
Signed-off-by: Pavan Kumar S <pavan.kumar.s@intel.com>
---
 drivers/usb/typec/usb_typec_detect.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/drivers/usb/typec/usb_typec_detect.c b/drivers/usb/typec/usb_typec_detect.c
index 257a1f0..0f141ff 100644
--- a/drivers/usb/typec/usb_typec_detect.c
+++ b/drivers/usb/typec/usb_typec_detect.c
@@ -199,6 +199,10 @@ static void detect_dfp_work(struct work_struct *work)
 			extcon_set_cable_state(detect->edev, "USB-Host", true);
 			atomic_notifier_call_chain(&detect->otg->notifier,
 				USB_EVENT_ID, NULL);
+			/* [WA] Since neccessary pmic-gpio isn't exposed
+			 * as of now, this is a w/a to write directly.
+			 */
+			intel_soc_pmic_writeb(0x6e2d, 0x31);
 
 			return;
 		} else if (CC_RA(cc1.v_rd) && CC_RA(cc2.v_rd)) {
@@ -412,6 +416,10 @@ static void update_phy_state(struct work_struct *work)
 							&cable_props);
 		} else {
 			/* state = DFP; disable VBUS */
+			/* [WA] Since neccessary pmic-gpio isn't exposed
+			 * as of now, this is a w/a to write directly.
+			 */
+			intel_soc_pmic_writeb(0x6e2d, 0x30);
 			extcon_set_cable_state(detect->edev,
 							"USB-Host", false);
 
-- 
1.9.1

